src/modules/mahjong/HandValidator/HandValidator.ts

```diff
@@ -0,0 +1,249 @@
+/**
+ * Class to store logic related to validating a Mahjong Hand
+ */
+import Tile from '../Tile/Tile';
+import { ValidPair, Meld, HandStructureResults } from '../types/MahjongTypes';
+import TileMapper from '../Tile/map/TileMapper';
+import SimpleTileTypes from '../enums/SimpleTileEnums';
+import MeldTypes from '../enums/MeldEnums';
+import sortHandUtils from '../utils/functions/sortHand';
+import Hand from '../Hand/Hand';
+
+class HandValidator {
+  public static TRIPLET_SIZE = 3;
+
+  public static KAN_SIZE = 4;
+
+  public static MIN_HAND_SIZE = 14;
+
+  public static DEFAULT_MELD_SIZE = 3;
+
+  public static REQUIRED_NUM_OF_MELDS = 4;
+
+  /**
+   * Creates an object mapping of the tiles to determine the number of each tile in the hand
+   * @param tiles An array of tile objects
+   * @returns object containing the number of each tile in the hand
+   */
+  public static createTileMapping(tiles: Tile[]): { [index: string]: number } {
+    const mapping: { [index: string]: number } = {};
+    const sortedTiles = sortHandUtils(tiles, Hand.generateHandWeights());
+    sortedTiles.forEach((t) => {
+      if (Object.prototype.hasOwnProperty.call(mapping, t.toString())) mapping[t.toString()] += 1;
+      else mapping[t.toString()] = 1;
```

Comment:
maybe simplify to:
```js
const tileStringDef = t.toString();
if (mapping.hasOwnProperty(tileStringDef)) mapping[tileStringDef] += 1;
else mapping[tileStringDef] = 1;
```
?

Reply:
can't do mapping.hasOwnProperty, but added tileStringdef

Reply:
ic, or you can use `in`, but I think `in` is a bit different from `hasOwnProperty`
```js
if (tileStringDef in mapping) mapping[tileStringDef] += 1;
```